Java/core/src/main/java/com/amazon/randomcutforest/PredictiveRandomCutForest.java

```diff
@@ -63,17 +77,28 @@ public class PredictiveRandomCutForest {
     protected ForestMode forestMode = ForestMode.STANDARD;
 
     public PredictiveRandomCutForest(Builder<?> builder) {
-        checkArgument(builder.forestMode != ForestMode.STREAMING_IMPUTE, "not suppoted yet");
         transformMethod = builder.transformMethod;
         Preprocessor.Builder<?> preprocessorBuilder = Preprocessor.builder().shingleSize(builder.shingleSize)
                 .transformMethod(builder.transformMethod).forestMode(builder.forestMode);
 
         int dimensions = builder.inputDimensions * builder.shingleSize;
         if (builder.forestMode == ForestMode.TIME_AUGMENTED) {
             dimensions += builder.shingleSize;
+            // if time is not differenced, then it can be added as a column
+            // without much difficulty
             preprocessorBuilder.normalizeTime(true);
             // force internal shingling for this option
             builder.internalShinglingEnabled = Optional.of(true);
+        } else if (builder.forestMode == ForestMode.STREAMING_IMPUTE) {
+            preprocessorBuilder.normalizeTime(true);
+            builder.internalShinglingEnabled = Optional.of(false);
```

Comment:
No - the invoking function will send inputLength size vectors. The shingling will be handled by the preprocessor but the RCF used internally will be externally shingled. The reason is that we/user may say "if we have shingle size X and have made up half the data then do not add the made up data to RCF' -- this is the imputedShingleFraction setting. This can be very helpful in stopping runaway feedback loops (hallucinations) since if the forecasted values are fed into the model we have a classic reinforcement scenario. This is the current state of ThresholdedRandomCutForest btw. One of the goals of this major refactor to make these options/capabities explicit.

Reply:
Not sure I understand. Will ask offline.

Reply:
yes, feel free to do it in the next PRs.

Reply:
Done. It wasn't as much -- we just edited those same lines for pauseSampling() and resumeSampling() for basic RCF (not TRCF -- because there is no good logic/decision mechanism to restart sampling for continuously evolving data once sampling is stopped). So the core of PointStore and the representations did not have to be touched. Of course we now have some redundant and deprecated function stubs -- those can be focus of subsequent PRs.